lms/services/grading_info.py

```diff
@@ -1,23 +1,13 @@
 from typing import List
 
-from marshmallow import fields
-
 from lms.models import GradingInfo, HUser
-from lms.validation import BasicLTILaunchSchema, ValidationError
 
 __all__ = ["GradingInfoService"]
 
 
 class GradingInfoService:
     """Methods for interacting with GradingInfo records."""
 
-    class _ParamsSchema(BasicLTILaunchSchema):
-        """Schema for the relevant parameters from the request."""
-
-        location = "form"
-        lis_result_sourcedid = fields.Str(required=True)
-        lis_outcome_service_url = fields.Str(required=True)
```

Comment:
I don't have an issue with using a schema (rather than direct Python code) to validate request params. But a schema like this definitely belongs in the views not in the service. Also, this looks like a schema for validating two fields in an HTML form submission, why is this a subclass of `BasicLTILaunchSchema`?

Reply:
Yeah, validating the request parameters using the schema is probably the right thing to do in our code base.

The odd thing here is doing it on a service which obscures the whole thing.